test/unittest/tests/wrappers/mock_session.cpp

```diff
@@ -28,11 +28,15 @@ mock_session::~mock_session()
 {
     WT_CONNECTION_IMPL *connection_impl = _mock_connection->get_wt_connection_impl();
     if (_session_impl->block_manager != nullptr)
-        __wt_free(nullptr, _session_impl->block_manager);
+        _session_impl->block_manager_cleanup(_session_impl);
     // FIXME-WT-13505: Move terminate function to connection once circular dependency is fixed.
     if (connection_impl->file_system != nullptr)
         utils::throw_if_non_zero(connection_impl->file_system->terminate(
           connection_impl->file_system, reinterpret_cast<WT_SESSION *>(_session_impl)));
+    if (_session_impl->dhandle->handle != nullptr)
```

Comment:
This doesn't look safe. We'll get a seg fault if the dhandle happens to be null.
```suggestion
    if (_session_impl->dhandle != nullptr && _session_impl->dhandle->handle != nullptr)
```

Reply:
This is what is causing the test failure @jiechenbo

Reply:
04ca1efd7a